crates/services/executor/src/executor.rs

```diff
@@ -916,9 +916,15 @@ where
                         if message.da_height() != da_height {
                             return Err(ExecutorError::RelayerGivesIncorrectMessages)
                         }
-                        block_storage_tx
-                            .storage_as_mut::<Messages>()
-                            .insert(message.nonce(), &message)?;
+                        let message_nonce = message.nonce();
+                        if !block_storage_tx
+                            .storage_as_ref::<Messages>()
+                            .contains_key(message_nonce)?
+                        {
+                            block_storage_tx
+                                .storage_as_mut::<Messages>()
+                                .insert(message_nonce, &message)?;
+                        }
                         execution_data
                             .events
                             .push(ExecutorEvent::MessageImported(message));
```

Comment:
should this also be conditional to the message_nonce not being already present in the block storage?

Reply:
Good catch. I've moved it.

This makes me wonder if we should be including an event in the case that a message is skipped. Or add it to a list of skipped messages, similar to the skipped tx.

Reply:
Added.